components/atom/switch/src/index.js

```diff
@@ -0,0 +1,193 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import AtomLabel from '@s-ui/react-atom-label'
+
+const BASE_CLASS = 'sui-AtomSwitch'
+
+const SIZES = {
+  MEDIUM: 'medium',
+  LARGE: 'large'
+}
+
+const TYPES = {
+  TOGGLE: 'toggle',
+  SELECT: 'select',
+  SINGLE: 'single'
+}
+
+class AtomSwitch extends Component {
+  constructor (props) {
+    super(props)
+
+    const {size, disabled, labelLeft, labelRight, type, name, label, labelOptionalText, onToggle} = props
+    this.onToggle = onToggle
+    this.size = size
+    this.type = type
+    this.labelRight = labelRight
+    this.labelLeft = labelLeft
+    this.name = name
+    this.label = label
+    this.labelOptionalText = labelOptionalText
+    this.state = {
+      toggle: false,
+      isFocus: false,
+      disabled
+    }
+
+    if (typeof window !== 'undefined' && window.addEventListener) {
+      window.addEventListener('keydown', this.keyBindings.bind(this), true)
+    }
+  }
+
+  keyBindings = (event) => {
+    const {isFocus} = this.state
+    if (!isFocus) {
+      return
+    }
+
+    if (event.defaultPrevented) {
+      return
+    }
+
+    switch (event.key) {
+      case 'Enter':
+        this.toggleSwitch()
+        event.preventDefault()
+        break
+      case ' ':
+        this.toggleSwitch()
+        event.preventDefault()
+        break
+    }
+  }
+
+  toggleSwitch = () => {
+    const {disabled, onToggle, toggle} = this.state
+    if (disabled) {
+      return
+    }
+    (onToggle) && onToggle(!toggle)
```

Comment:
I would say to put as a defaultProp an empty method rather than checking if the `onToggle` is a function.